tests/e2e-inst-test.sh

```diff
@@ -130,202 +235,361 @@ lsm_test_not_supported=0
+declare -A test_pids_map=() # Map: test_name -> pid
+max_internal_sleep=0
+max_test_timeout=0
 
-sleep 3
+for TEST in ${TESTS}; do
+    print_test_header "${TEST}" "RUNNING"
 
-info "= RUNNING TESTS ================================================"
-info
-# Run tests, one by one
+    # Check for skip conditions first
+    if should_skip_test "${TEST}"; then
+        info "skipping ${TEST}"
+        print_end_separator
+        continue
+    fi
 
-for TEST in $TESTS; do
+    test_timeout=$(get_test_timeout "${TEST}")
+    test_sleep=$(get_test_sleep "${TEST}")
+    export E2E_INST_TEST_SLEEP="${test_sleep}" # test will use this env variable for sleeping
 
-    info
-    info "= TEST: $TEST =============================================="
-    info
+    # Track maximum internal sleep time and timeout
+    if [[ "${test_sleep}" -gt "${max_internal_sleep}" ]]; then
+        max_internal_sleep="${test_sleep}"
+    fi
+    if [[ "${test_timeout}" -gt "${max_test_timeout}" ]]; then
+        max_test_timeout="${test_timeout}"
+    fi
 
-    case $TEST in
-    HOOKED_SYSCALL)
-        # wait for tracee hooked event to be processed
-        info "waiting for tracee hooked event to be processed"
-        sleep 15
-        ;;
-    FTRACE_HOOK)
-        info "waiting for tracee ftrace hook event to be processed"
-        sleep 15
-        ;;
-    *)
-        info "running test $TEST"
-        timeout --preserve-status $TRACEE_RUN_TIMEOUT "${TESTS_DIR}"/"${TEST,,}".sh
-        ;;
+    # Set test arguments and run test
+    test_args=""
+    case ${TEST} in
+        HOOKED_SYSCALL)
+            info "unloading hijack module that was loaded at setup step"
+            test_args="--uninstall"
+            ;;
+        FTRACE_HOOK)
+            info "loading and unloading ftrace hook module"
+            test_args="--install --uninstall"
+            ;;
+        *)
+            case ${TEST} in
+                CONTAINERS_DATA_SOURCE | \
+                    BPF_ATTACH | \
+                    SUSPICIOUS_SYSCALL_SOURCE | \
+                    STACK_PIVOT | \
+                    WRITABLE_DATA_SOURCE | \
+                    SECURITY_PATH_NOTIFY | \
+                    PROCTREE_DATA_SOURCE | \
+                    VFS_WRITE)
+                    test_args="--run"
+                    ;;
+            esac
+
+            run_msg="running test ${TEST} with timeout ${test_timeout}"
+            if [[ "${test_sleep}" -gt 0 ]]; then
+                run_msg="${run_msg} and internal sleep ${test_sleep}"
+            fi
+            info "${run_msg}"
+            ;;
     esac
+
+    # Run test in background and store PID in map
+    # Redirect output to file to show later during wait phase
+    test_output_file="/tmp/test_${TEST,,}_$$"
+    (
+        # shellcheck disable=SC2086
+        timeout --preserve-status "${test_timeout}" "${TESTS_DIR}"/"${TEST,,}".sh ${test_args:-}
+    ) > "${test_output_file}" 2>&1 &
+    test_pids_map["${TEST}"]=$!
```

Comment:
As the sleep (E2E_INST_TEST_SLEEP) is being passed via export, it's crucial to check if running tests in parallel cannot cause race conditions. Perhaps just changing the command, appending the env var, might solve the problem